atproto/lexicon/catalog.go

```diff
@@ -72,46 +76,53 @@ func (c *BaseCatalog) AddSchemaFile(sf SchemaFile) error {
 			return err
 		}
 		s := Schema{
-			ID:       name,
-			Revision: sf.Revision,
-			Def:      def.Inner,
+			ID:  name,
+			Def: def.Inner,
 		}
 		c.schemas[name] = s
 	}
 	return nil
 }
 
+// internal helper for loading file paths (either real filesystem or embed.FS)
+func (c *BaseCatalog) addDirEntry(p string, d fs.DirEntry, err error) error {
+	if err != nil {
+		return err
+	}
+	if d.IsDir() {
+		return nil
+	}
+	if !strings.HasSuffix(p, ".json") {
+		return nil
+	}
+	slog.Debug("loading Lexicon schema file", "path", p)
+	f, err := os.Open(p)
```

Comment:
`os.Open()` won't work for the LoadEmbedFS() case

Reply:
ok, i'll add a test to cover that use-case

Reply:
think I got this fixed. thanks for the catch!